power_grid_model_c/power_grid_model/include/power_grid_model/optimizer/tap_position_optimizer.hpp

```diff
@@ -0,0 +1,86 @@
+// SPDX-FileCopyrightText: Contributors to the Power Grid Model project <powergridmodel@lfenergy.org>
+//
+// SPDX-License-Identifier: MPL-2.0
+
+#pragma once
+
+#include "base_optimizer.hpp"
+
+#include "../auxiliary/dataset.hpp"
+#include "../common/enum.hpp"
+
+#include <boost/graph/compressed_sparse_row_graph.hpp>
+
+namespace power_grid_model::optimizer {
+namespace tap_position_optimizer {
+
+namespace detail = power_grid_model::optimizer::detail;
+
+using GraphIdx = size_t;
+struct Edge {
+    Idx2D pos{};
+    Idx weight{};
+};
```

Comment:
`TrafoGraphIdx` and `TrafoGraphEdge`? Or if a very general index and `edge` concept in graph sense, it should reside somewhere in `core`.

Reply:
you can do that in your ticket. i deem that out of scope of this PR